src/main/kotlin/com/openlattice/data/storage/PostgresQueries.kt

```diff
@@ -127,17 +127,17 @@ internal fun buildWithClause(queryId: UUID, entityKeyIds: Map<UUID, Optional<Set
     val entitiesJoinCondition = buildEntitiesJoinCondition(IDS.name, linking)
     val linkingClause = if (linking) "AND ${LINKING_ID.name} IS NOT NULL" else ""
 
-    val emptyEntitySetIds = entityKeyIds.asSequence().filter { it.value.isEmpty }
-    val completeEntitySetsUnionSql = if (emptyEntitySetIds.iterator().hasNext()) { // if there are entity set ids with no entity ids in entityKeyIds
-        val completeEntitySets = emptyEntitySetIds.joinToString(",") { "'${it.key}'" }
-        "UNION SELECT $selectColumns FROM ${IDS.name} WHERE ${ENTITY_SET_ID.name} IN ($completeEntitySets)"
-    } else {
-        ""
-    }
+    val dataKeySql = "AND " + entityKeyIds.map { (entitySetId, entityKeyIds) ->
```

Comment:
Can we put this back into buildEntitiesClause? Or just uncomment that part and use it here?

Reply:
Once we make sure those queries aren't causing perf issues.

Reply:
Why not? I brought it back.